robot-server/tests/sessions/test_sessions_router.py

```diff
@@ -0,0 +1,337 @@
+) -> None:
+    """It should return an empty collection response with no sessions exist."""
+    decoy.when(session_store.get_all()).then_return([])
+
+    response = client.get("/sessions")
+
+    verify_response(response, expected_status=200, expected_data=[])
+
+
+def test_get_sessions_not_empty(
+    decoy: Decoy,
+    session_builder: SessionBuilder,
+    session_store: SessionStore,
+    client: TestClient,
+) -> None:
+    """It should return an empty collection response with no sessions exist."""
+    created_at_1 = datetime.now()
+    created_at_2 = datetime.now()
+
+    entry_1 = SessionStoreEntry(
+        session_id="unique-id-1", session_data=None, created_at=created_at_1, inputs=[]
+    )
+    entry_2 = SessionStoreEntry(
+        session_id="unique-id-2", session_data=None, created_at=created_at_2, inputs=[]
+    )
+
+    session_1 = BasicSession(id="unique-id-1", createdAt=created_at_1, inputs=[])
+    session_2 = BasicSession(id="unique-id-2", createdAt=created_at_2, inputs=[])
+
+    decoy.when(session_store.get_all()).then_return([entry_1, entry_2])
+
+    decoy.when(
+        session_builder.build(
+            session_id="unique-id-1",
+            session_data=None,
+            created_at=created_at_1,
+            inputs=[],
+        )
+    ).then_return(session_1)
+
+    decoy.when(
+        session_builder.build(
+            session_id="unique-id-2",
+            session_data=None,
+            created_at=created_at_2,
+            inputs=[],
+        )
+    ).then_return(session_2)
+
+    response = client.get("/sessions")
+
+    verify_response(response, expected_status=200, expected_data=[session_1, session_2])
+
+
+def test_delete_session_by_id(
+    decoy: Decoy,
+    session_store: SessionStore,
+    client: TestClient,
+) -> None:
+    """It should be able to remove a session by ID."""
+    response = client.delete("/sessions/unique-id")
+
+    decoy.verify(session_store.remove(session_id="unique-id"))
+
+    assert response.status_code == 200
+    assert response.json()["data"] is None
+
+
+def test_delete_session_with_bad_id(
+    decoy: Decoy,
+    session_store: SessionStore,
+    client: TestClient,
+) -> None:
+    """It should 404 if the session ID does not exist."""
+    key_error = SessionNotFoundError(session_id="session-id")
+
+    decoy.when(session_store.remove(session_id="session-id")).then_raise(key_error)
+
+    response = client.delete("/sessions/session-id")
+
+    verify_response(
+        response,
+        expected_status=404,
+        expected_errors=SessionNotFound(detail=str(key_error)),
+    )
+
+
+def test_create_session_input(
+    decoy: Decoy,
+    session_store: SessionStore,
+    session_runner: SessionRunner,
+    unique_id: str,
+    current_time: datetime,
+    client: TestClient,
+) -> None:
+    """It should handle a pause input."""
+    expected_input = SessionInput(
+        inputType=SessionInputType.START,
+        createdAt=current_time,
+        id=unique_id,
+    )
```

Comment:
Notwithstanding my questioning of whether a "start input" is something that we should have:

```suggestion
    """It should handle a start input."""
    expected_input = SessionInput(
        inputType=SessionInputType.START,
        createdAt=current_time,
        id=unique_id,
    )
```

Typo?